### PR TITLE
Add Omega docs link to index

### PR DESCRIPTION
**Pull request by altheaden**
Add a link to the Omega documentation to `docs/index.md`.

### CODE REVIEW

**Comment by xylar** [comment]
@mahf708, I'm guessing this is no problem but wanted to check with you.  The link is less of a question, it's more the interaction between CI on E3SM (https://github.com/E3SM-Project/E3SM/blob/master/.github/workflows/e3sm-gh-pages.yml) and on Omega (https://github.com/E3SM-Project/Omega/blob/develop/.github/workflows/docs_workflow.yml).  Any concerns?

BTW, I will ask @altheaden to make a PR to rename the 2 Omega workflows under https://github.com/E3SM-Project/Omega/blob/develop/.github/workflows/ to have an `omega_` prefix.

**Review by xylar — APPROVED** [review]

**Comment by xylar** [comment]
@jonbob, I don't know if you want to be the integrator here or to reassign to someone else (e.g. @rljacob).

**Comment by xylar** [comment]
@mahf708, yes we're aware that some planning will presumably be needed when we merge Omega back into E3SM/master, including making sure this CI works as expected.  At that point, we will need to decide if the documentation link should now be to the (presumably more stable) E3SM version of the Omega docs or the one on the Omega fork.

File: docs/index.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ research problems and Department of Energy mission needs while efficiently using
 - [MOSART](./MOSART/index.md)
 - [MPAS-Ocean](./MPAS-Ocean/index.md)
 - [MPAS-seaice](./MPAS-seaice/index.md)
+- [Omega](https://docs.e3sm.org/Omega/omega/)
```
[the remark this answers is not in the document]
Comment (xylar) [reply docs/index.md:26]: @mahf708, yes to "not yet supported" but I don't think we are writing Omega with all caps and that should be fixed in its own docs.

[the remark this answers is not in the document]
Comment (xylar) [reply docs/index.md:26]:
```suggestion
- [Omega](https://docs.e3sm.org/Omega/omega/)  - not yet supported.
```

[the remark this answers is not in the document]
Comment (xylar) [reply docs/index.md:26]: I've added a task for @altheaden to make the docs consistently Omega, not OMEGA.

[the remark this answers is not in the document]
Comment (rljacob) [reply docs/index.md:26]: Isn't OMEGA an acronym?  Seems like it should be all caps everywhere.  What does @vanroekel say?

[the remark this answers is not in the document]
Comment (vanroekel) [reply docs/index.md]: @rljacob We've discussed this within the Omega team and would like to stick with Omega as it is closer to the current software stylistic preferences.

**Review by rljacob — APPROVED** [review]

**Comment by xylar** [comment]
@jonbob, I think this is ready to merge when you're happy.

**Comment by jonbob** [comment]
@rljacob -- I'm not sure how to test documentation so I moved this to you

**Pull request merged**